homeassistant/components/sia/config_flow.py

```diff
@@ -0,0 +1,134 @@
+"""Config flow for sia integration."""
+from pysiaalarm import (
+    InvalidAccountFormatError,
+    InvalidAccountLengthError,
+    InvalidKeyFormatError,
+    InvalidKeyLengthError,
+    SIAAccount,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_PORT
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_ADDITIONAL_ACCOUNTS,
+    CONF_ENCRYPTION_KEY,
+    CONF_PING_INTERVAL,
+    CONF_ZONES,
+    DOMAIN,
+)
+
+HUB_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PORT): int,
+        vol.Required(CONF_ACCOUNT): str,
+        vol.Optional(CONF_ENCRYPTION_KEY, None): str,
+        vol.Required(CONF_PING_INTERVAL): int,
+        vol.Required(CONF_ZONES, default=1): int,
+        vol.Optional(CONF_ADDITIONAL_ACCOUNTS, default=False): bool,
+    }
+)
+
+ACCOUNT_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ACCOUNT): str,
+        vol.Optional(CONF_ENCRYPTION_KEY, None): str,
+        vol.Required(CONF_PING_INTERVAL): int,
+        vol.Required(CONF_ZONES, default=1): int,
+        vol.Optional(CONF_ADDITIONAL_ACCOUNTS, default=False): bool,
+    }
+)
+
+
+def validate_input(data: dict) -> bool:
+    """Validate the input by the user."""
+    SIAAccount(data[CONF_ACCOUNT], data.get(CONF_ENCRYPTION_KEY))
+
+    ping = int(data[CONF_PING_INTERVAL])
+    if not 1 <= ping <= 1440:
+        raise InvalidPing
+    zones = int(data[CONF_ZONES])
+    if zones <= 0:
+        raise InvalidZones
+
+
+class SIAConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for sia."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+    data = {}
+
+    async def async_step_user(self, user_input: dict = None):
+        """Handle the initial step."""
+        errors = {}
+        if not user_input:
```

Comment:
You can reverse the condition, and only call `async_show_form` from one location

Reply:
I decided to follow the patterns of reducing indents and return asap, cleaned up the whole config quite a bit based also on the other comment about a extra step. So that is much easier to read now.